VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+
+An example version string for a UXP release that embeds Crossplane `v1.3.0`
+could be:
+
+```
+v1.3.0-up.2
+```
+
+Additional pre-release segments may be appended both before and after the
+`-up-<components>` segment. Any pre-release segments appended before will
+correspond to the embedded Crossplane version (e.g. `v1.4.0-rc.0-up.1`). Any
+pre-release segments appended after will correspond to a pre-release version of
+UXP itself (e.g. `v1.4.0-up.1-rc.0`). There should _never_ be both a Crossplane
+pre-release and UXP pre-release on the same version string (see [Guarantees
+section](#guarantees)).
+
+## Terms
+
+The following terms are used throughout this document and are defined as
+follows:
+
+- **UXP Version**: the entire semantic version for a UXP release (e.g.
+  `v1.2.2-up.3`).
+- **Crossplane Version**: the version of Crossplane embedded in the UXP release
+  (e.g. `v1.2.2`).
+- **Components Marker**: the version of the components surrounding upstream (or
+  temporarily forked) Crossplane (e.g. `-up.3`).
+
+## Guarantees
+
+![versioning-diagram](docs/media/versioning.png)
+
+The UXP versioning strategy can be expressed through a series of guarantees,
+each of which are elaborated in the subsequent sections. The diagram above
+illustrates examples of each guarantee in practice.
+
+1. The components marker on UXP main should always end in `-up.1` (i.e. no
+   components release candidates).
+2. The Crossplane version on UXP main must always be a release candidate version
+   for the first release of a major/minor pair (i.e. `vX.Y.0-rc.Z`).
+3. The Crossplane version on any `release-*` branch should _always_ be a
+   Crossplane release version (e.g. `v1.2.2`) (i.e. no release candidates).
+4. Not every Crossplane patch for a given `<major>`/`<minor>` version must be
+   included in a UXP release, even if the `<major>`/`<minor>` version has been
+   selected for long-term service (LTS). This would typically occur if
+   Crossplane made multiple patch releases in quick succession and a UXP version
+   had not been released for each in the intermediate (e.g. if `v1.2.4` is
+   released in upstream Crossplane and has not been included in a UXP release
+   before `v1.2.5` is released upstream, UXP may elect to forego cutting a
+   release with `v1.2.4`).
+5. At least one release candidate (RC) version should exist for any version that
+   is released. This is typically accomplished by cutting the RC version for the
+   _next_ release immediately after release of the preceding version.
+6. As a consequence of the preceding rule, a UXP version on a `release-*` branch
+   may end in `-rc.X` (e.g. `v1.2.2-up.4-rc.0`).
+7. A release _always_ results in the subsequent tag on the same branch to have
+   an incremented components version (e.g. the tag after `v1.2.2-up.2` must
+   include `-up.3`).
+8. A release for an embedded Crossplane version that has not yet been included
+   in a UXP release must have a components marker of `-up.1`.
```

Comment:
`rc-` starts with `0` but `up-` starts with `1`. Could we follow the same for both?

Reply:
I would be in favor of that, but I was basing this off of https://github.com/upbound/universal-crossplane/blob/main/design/design-packaging.md#versioning, in which I feel like there was a reason for starting with `-up.1`, but I cannot remember why, do you happen to know?

Reply:
No, I don't remember either. I believe we can start with `-up.0` similar to how a patch starts with 0, e.g. `1.3.0`.